### PR TITLE
Update Tuweni dependencies to 2.0.0

### LINKED ISSUE #3419: Upgrade to Tuweni libs 1.4.0 when released
Tuweni `1.4.0` release is coming soon and there would be some fixes in `Bytes` library, so it makes sense to upgrade

https://github.com/apache/incubator-tuweni/releases

### PR DESCRIPTION
**Pull request by ajsutton**
## PR Description
Updates Tuweni dependencies to version 2.0.0.

fixes #3419

## Documentation

- [x] I thought about documentation and added the `documentation` label to this PR if updates are required.

## Changelog

- [x] I thought about adding a changelog entry, and added one if I deemed necessary.

### CODE REVIEW

File: gradle/license-report-config/allowed-licenses.json [hunk 1]
```diff
@@ -66,6 +66,9 @@
     },
     {
       "moduleName": "io.libp2p:jvm-libp2p-minimal"
+    },
+    {
+      "moduleName": "org.jetbrains.kotlinx:kotlinx-coroutines-core"
```
Comment (tbenr) [inline gradle/license-report-config/allowed-licenses.json:71]: are these license additions related to the PR?

**Review by tbenr — APPROVED** [review]
LGTM with a question

File: gradle/license-report-config/allowed-licenses.json (hunk 1 above)
Comment (ajsutton) [reply gradle/license-report-config/allowed-licenses.json:71]: Yes, it appears to be a new or updated dependency from Tuweni.

I'm going to switch this back to a draft PR though as I've reproduced the performance problem we originally saw in #3419. I can work around the worst of it by removing our use of tuweni-crypto but I think we're going to need a new Tuweni release to fix it properly as things like tuweni-ssz are still using tuweni-crypto and it's now incredibly slow.

**Comment by ajsutton** [comment]
Closing this - will evaluate Tuweni 2.0.1 to see if it fixes the performance regression in Hash since it's still used in a few places internal to Tuweni.
